.github/workflows/sanity.yml

```diff
@@ -127,3 +128,33 @@ jobs:
           make deploy
           kubectl get deployment -n ramen-system
           kubectl -n ramen-system wait --for=condition=Available --timeout=60s deploy/ramen-controller-manager
+
+  publish-image:
+    name: Publish built image
+    needs: [deploy-check, lint, golangci]
```

Comment:
Few items in this are not required? i assume PR will get merge only when these tests are passed.

Reply:
True, but job runs post a push as well, so say there was any auto merge etc. then re-running the jobs and then on success pushing it is safer.